synapse/lib/cell.py

```diff
@@ -1833,75 +1834,89 @@ def getproc():
 
     async def iterNewBackupArchive(self, user, name=None, remove=False):
 
-        success = False
-        loglevel = logging.WARNING
+        if self.backupstreaming:
+            raise s_exc.BackupAlreadyRunning(mesg='Another streaming backup is already running')
 
-        if name is None:
-            name = time.strftime('%Y%m%d%H%M%S', datetime.datetime.now().timetuple())
+        try:
+            if remove:
+                self.backupstreaming = True
 
-        path = self._reqBackDirn(name)
-        if os.path.isdir(path):
-            mesg = 'Backup with name already exists'
-            raise s_exc.BadArg(mesg=mesg)
+            success = False
+            loglevel = logging.WARNING
 
-        link = s_scope.get('link')
-        linkinfo = await link.getSpawnInfo()
-        linkinfo['logconf'] = await self._getSpawnLogConf()
+            if name is None:
+                name = time.strftime('%Y%m%d%H%M%S', datetime.datetime.now().timetuple())
 
-        try:
-            await self.runBackup(name)
-        except Exception:
-            if remove:
-                logger.debug(f'Removing {path}')
-                await s_coro.executor(shutil.rmtree, path, ignore_errors=True)
-                logger.debug(f'Removed {path}')
-            raise
+            path = self._reqBackDirn(name)
+            if os.path.isdir(path):
+                mesg = 'Backup with name already exists'
+                raise s_exc.BadArg(mesg=mesg)
 
-        await self.boss.promote('backup:stream', user=user, info={'name': name})
+            link = s_scope.get('link')
+            linkinfo = await link.getSpawnInfo()
+            linkinfo['logconf'] = await self._getSpawnLogConf()
 
-        ctx = multiprocessing.get_context('spawn')
+            try:
+                await self.runBackup(name)
+            except Exception:
+                if remove:
+                    logger.debug(f'Removing {path}')
+                    await s_coro.executor(shutil.rmtree, path, ignore_errors=True)
+                    logger.debug(f'Removed {path}')
+                raise
 
-        proc = None
-        mesg = 'Streaming complete'
+            await self.boss.promote('backup:stream', user=user, info={'name': name})
 
-        def getproc():
-            proc = ctx.Process(target=_iterBackupProc, args=(path, linkinfo))
-            proc.start()
-            return proc
+            ctx = multiprocessing.get_context('spawn')
 
-        try:
-            proc = await s_coro.executor(getproc)
+            proc = None
+            mesg = 'Streaming complete'
 
-            await s_coro.executor(proc.join)
+            def getproc():
+                proc = ctx.Process(target=_iterBackupProc, args=(path, linkinfo))
+                proc.start()
+                return proc
 
-        except (asyncio.CancelledError, Exception) as e:
+            try:
+                if remove:
+                    self.backupstreaming = True
```

Comment:
we're setting this variable at the top and middle of the function ?

Reply:
this shouldn't be here, will remove